src/OrchardCore.Modules/OrchardCore.ContentTypes/RecipeSteps/ContentDefinitionStep.cs

```diff
@@ -58,6 +58,11 @@ private Task UpdateContentTypeAsync(ContentTypeDefinition type, ContentTypeDefin
 
                 foreach (var part in record.ContentTypePartDefinitionRecords)
                 {
+                    if (string.IsNullOrEmpty(part.PartName))
+                    {
+                        throw new InvalidOperationException($"Unable to add content-part to the '{type.Name}' content-type. The part name cannot be null or empty.");
```

Comment:
Why not `ArgumentNullException` or `ArgumentException`?

Reply:
Uh, I'm not sure, here's a suggestion from @Piedone
It is possible that there is not a problem with the function's arguments here, indicating that the program is performing an unsupported operation

Reply:
Maybe because it's not passed in, no problem